tls/s2n_handshake_io.c

```diff
@@ -483,6 +494,11 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     /* In the initial TLS1.3 release, we will only support the basic handshake. */
     if (IS_TLS13_HANDSHAKE(conn)) {
         conn->handshake.handshake_type |= FULL_HANDSHAKE;
+
+        if (conn->handshake.hello_retry_request == 1) {
+            conn->handshake.handshake_type |= HELLO_RETRY_REQUEST;
+        }
```

Comment:
Do we need a separate flag? Can we set the handshake_type directly from s2n_set_hello_retry_required?

Reply:
This has been updated to remove the state from the handshake struct.